### PR TITLE
Add Tasks Documentation

### PR DESCRIPTION
**Pull request by ncapps**
This change adds content to the **Core Tasks** section of the new documentation site.

Resolves: https://github.com/kubernetes-sigs/kustomize/issues/5286

### CODE REVIEW

File: site/content/en/docs/Tasks/configmaps_and_secrets.md [hunk 1]
```diff
@@ -1,8 +1,51 @@
 ---
 title: "ConfigMaps and Secrets"
 linkTitle: "ConfigMaps and Secrets"
-weight: 3
+weight: 1
 date: 2017-01-05
 description: >
```
Comment (bugoverdose) [inline site/content/en/docs/Tasks/configmaps_and_secrets.md:6]: I suggest updating the dates to September 2023. 

Of course, I think we need a clear guideline on updating the dates.

File: site/content/en/docs/Tasks/labels_and_annotations.md [hunk 2]
```diff
@@ -1,8 +1,46 @@
...
+---
+
+{{< alert color="success" title="TL;DR" >}}
+- Set Labels for all Resources declared within a project with `commonLabels`
+- Set Annotations for all Resources declared within a project with `commonAnnotations`
+{{< /alert >}}
+
+## Motivation
+Users may want to define a common set of labels or annotations for all the Resources in a project.
+
```
Comment (bugoverdose) [inline site/content/en/docs/Tasks/labels_and_annotations.md:17]: In this sentence, the doc is referring to the users as if they are third party.
So it sort of makes as feel like this doc is for the contributors to read, not the actual users of kustomize users. Same goes for all the other first sentences of **Motivation**.

```suggestion
It's possible to define a common set of labels or annotations for all the Resources in a project.
```

```suggestion
You can define a common set of labels or annotations for all the Resources in a project.
```

File: site/content/en/docs/Tasks/configmaps_and_secrets.md [hunk 3]
```diff
@@ -1,8 +1,51 @@
...
\ No newline at end of file
+---
+
+{{< alert color="success" title="TL;DR" >}}
+- Generate ConfigMaps from files and literals with `configMapGenerator`
+- Generate Secrets from files and literals with `secretGenerator`
+- Rollout ConfigMap and Secret changes to workloads
+{{< /alert >}}
+
+## Motivation
```
Comment (bugoverdose) [inline site/content/en/docs/Tasks/configmaps_and_secrets.md:16]: I don't think the section name **Motivation** is necessary.

```suggestion

```

**Review by bugoverdose — COMMENTED** [review]
I'm leaving just a few minor suggestions. Good work though.

**Comment by natasha41575** [comment]
I think the tests aren't triggering because of https://github.com/kubernetes-sigs/kustomize/pull/5332, I'm going to revert that one today.

File: site/content/en/docs/Tasks/configmaps_and_secrets.md (hunk 1 above)
Comment (ncapps) [reply site/content/en/docs/Tasks/configmaps_and_secrets.md:6]: I updated this to the current date for all task files.

File: site/content/en/docs/Tasks/configmaps_and_secrets.md (hunk 3 above)
Comment (ncapps) [reply site/content/en/docs/Tasks/configmaps_and_secrets.md:16]: The **Motivation** header was copied from the original site [here](https://kubectl.docs.kubernetes.io/guides/config_management/secrets_configmaps/). All other Task pages include a Motivation section as an introduction to the concept.

Personally, I think it makes sense to keep this header but I can remove if you feel strongly that it is not helpful.

**Comment by annasong20** [comment]
@ncapps Thanks so much for working on this section! You did exactly as the [sheet](https://docs.google.com/spreadsheets/d/1Aaal0Ou10xeDkh_COWkg_r00R0I-Il_t5G2ad6W3jWc/edit#gid=0) instructed, which was to leave only the conceptual information in the "core_tasks" section.

Unfortunately, I didn't write this sheet, and didn't catch these comments. Sorry in advance. I found that the structure of the [Tasks](https://kubernetes.io/docs/tasks/) section in the Kubernetes documentation belies the excel sheet comments in that their Tasks section only documents concrete examples or as they call it, "a short sequence of steps". I wanted to discuss with you what our Tasks section should hold. If we follow the Kubernetes example, we can discuss on this PR where to move the content or work to merge this PR with the assumption that we'll move it in the future.

**Comment by ncapps** [comment]
Hi @annasong20, Thanks for taking a look at my PR. Please let me know what you envision for the Kustomize Tasks section and I will update accordingly with this PR.

File: site/content/en/docs/Tasks/configmaps_and_secrets.md (hunk 3 above)
Comment (bugoverdose) [reply site/content/en/docs/Tasks/configmaps_and_secrets.md:16]: I think it's OK to keep the header if you think it's useful.

**Comment by annasong20** [comment]
Hi @ncapps, sorry for the delayed response. Also feel free to DM me on Slack if that's faster.

I don't have a great vision of all the sub-sections, but my reasoning was that in the absence of strong opinions, our site might be easier to understand if it follows the same format as [Kubernetes](https://kubernetes.io/docs/home/). Their [Tasks](https://kubernetes.io/docs/tasks/) section seems to feature concrete steps for accomplishing simple workflows, aka Tasks. Taking `configMapGenerator` as an example,

* the configuration [example](https://kubectl.docs.kubernetes.io/guides/config_management/secrets_configmaps/#configmapsgenerator) with `.properties` currently under `Guides`
* all the different [examples](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/configmapgenerator/#configmap-from-file) of the different sub-fields under `Reference`

seem like they belong under `Tasks`. The [`Reference`](https://kubernetes.io/docs/reference/) looks like it should be used for quick lookups; I made the same belated [comment](https://github.com/kubernetes-sigs/kustomize/pull/5313) on @bugoverdose's PR. For example, basic field [syntax](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/) instead of containing examples. [Motivation](https://kubectl.docs.kubernetes.io/guides/config_management/secrets_configmaps/#motivation), introduction, background, reasoning, etc. seem like they should belong under [Concepts](https://kubernetes.io/docs/concepts/), where I'd probably look to "understand" a topic.

I also think we might benefit from working on 1 or a few fields at a time. We can finalize our approach at a smaller scope and then you can apply it to all other fields. This way, we don't waste your time applying dissented changes between reviews.

Disclaimer that my take is not authoritative. Feel free to push back where you disagree! Treating this more like a collaboration.

**Comment by ncapps** [comment]
I am closing the PR. Feedback will be addressed in https://github.com/kubernetes-sigs/kustomize/pull/5368 and https://github.com/kubernetes-sigs/kustomize/pull/5383.

**Pull request closed without merging**